app_dart/lib/src/foundation/github_checks_util.dart

```diff
@@ -74,14 +74,24 @@ class GithubChecksUtil {
   }
 
   Future<github.CheckRun> getCheckRun(
-    github.GitHub gitHubClient,
+    Config cocoonConfig,
```

Comment:
Do we need tests to validate the retries are actually happening?

Reply:
Have searched around, but didn't find an easy way to mock the corresponding function (`getCheckRun`), which is located in a private class...  Not sure if you got any chance to look at this when implementing other functions like `updateCheckRun`?
If not, I may prefer to skip it here as I have spent quite a while on this, and this is not a blocking thing.

Reply:
Another potential blocking thing on the test is: we need to return different values for the same mock call, but it is not supported by default via dart mockito (though Java supports that by default). It seems doable if we return same type of values (https://github.com/dart-lang/mockito/issues/260), but here we need to return a combination of an exception and a valid value.